bazel/packaging/packaging.bzl

```diff
@@ -358,3 +358,154 @@ redpanda_package = rule(
     },
     toolchains = ["@bazel_tools//tools/cpp:toolchain_type"],
 )
+
+def _prepapare_package_conent(ctx):
+    cc_binaries = [
+    ]
+    for b in ctx.attr.cc_binaries:
+        for file in b.files.to_list():
+            cc_binaries.append(struct(attr = b, file = file))
+    install_path_value = ctx.attr.install_path[BuildSettingInfo].value
+    package_cc_binaries = _prepare_package_binaries(
+        ctx,
+        cc_binaries,
+        "{}/lib".format(install_path_value),
+    )
+
+    binaries = []
+    for b in ctx.attr.binaries:
+        for file in b.files.to_list():
+            binaries.append(file)
+
+    return struct(
+        cc_binaries = package_cc_binaries.binary_files,
+        binaries = binaries,
+        shared_libraries = package_cc_binaries.shared_libraries,
+    )
+
+def _create_directory_cfg(ctx, package_content):
+    package_files = []
+    for b in package_content.binaries:
+        package_files.append({
+            "path": "bin",
+            "name": b.basename,
+            "source": b.path,
+        })
+
+    # in directory we put cc_binaries into libexec so they can be executed with a script that is created later
+    for b in package_content.cc_binaries:
+        package_files.append({
+            "path": "bin",
+            "name": b.basename.removesuffix("_ld").removesuffix("_patched"),
+            "source": b.path,
+        })
+
+    for sl in package_content.shared_libraries:
+        package_files.append({
+            "path": "lib",
+            "name": sl.basename,
+            "source": sl.path,
+        })
+
+    return {
+        "package_dirs": ["bin", "lib"],
+        "directory_mode": True,
+        "owner": ctx.attr.owner,  # Default owner
+        "package_files": package_files,
+    }
+
+def _create_tarball_cfg(ctx, package_content):
```

Comment:
This looks like the same as the directory one, save the bool flag. Can we unify?